xarray/core/alignment.py

```diff
@@ -710,7 +745,8 @@ def align(
         - "override": if indexes are of same size, rewrite indexes to be
           those of the first object with that dimension. Indexes for the same
           dimension must have the same size in all objects.
-
+    broadcast : bool
+        The alignment fails if dimensions' names differ.
```

Comment:
```suggestion
    broadcast : bool, optional
        Turn off automatic broadcasting of all objects along dimensions that are present in some but not all objects.
        If False, this will raise an error when all objects do *not* have the same dimensions.
```